src/nvidia_rag/ingestor_server/server.py

```
@@ -192,6 +317,16 @@ class DocumentUploadRequest(BaseModel):
         exclude=True,  # WAR to hide it from openapi schema
     )
 
+    @model_validator(mode="after")
+    def validate_summary_configuration(self) -> "DocumentUploadRequest":
+        """Validate that summary_options is only used when generate_summary is True."""
+        if self.summary_options and not self.generate_summary:
+            raise ValueError(
+                "summary_options can only be provided when generate_summary=True. "
+                "Either set generate_summary=True or remove summary_options."
+            )
```

Ruff (0.14.5)

Avoid specifying long messages outside the exception class

(TRY003)